src/main.rs

```diff
@@ -365,114 +629,122 @@ fn main() {
                     sleep(Duration::from_millis(1000));
                     run_pin.set_high();
                 }
-                match SocketStream::from_address(&address, *&config.use_wifi) {
-                    Ok(mut stream) => {
-                        info!("Connected to {}, waiting for frames from rp2040", if config.use_wifi { &"tc2-frames server"} else { &"thermal-recorder unix socket"});
-                        let mut sent_header = false;
-                        let mut ms_elapsed = 0;
-                        'send_loop: loop {
-                            // Check if we need to reset rp2040 because of a config change
-                            if let Ok(_) = restart_rx.try_recv() {
-                                cross_thread_signal_2.store(true, Ordering::Relaxed);
-                                loop {
-                                    info!("Restarting rp2040");
-                                    if !run_pin.is_set_high() {
-                                        run_pin.set_high();
-                                        sleep(Duration::from_millis(1000));
+                if frame_acquire {
+                    info!("Connecting to frame socket {}", address);
+
+                    match SocketStream::from_address(&address, *&config.use_wifi) {
+                        Ok(mut stream) => {
+                            info!("Connected to {}, waiting for frames from rp2040", if config.use_wifi { &"tc2-frames server"} else { &"thermal-recorder unix socket"});
+                            let mut sent_header = false;
+                            let mut ms_elapsed = 0;
+                            'send_loop: loop {
+                                // Check if we need to reset rp2040 because of a config change
+                                if let Ok(_) = restart_rx.try_recv() {
+                                    if let Ok(config) = DeviceConfig::load_from_fs() {
```

Comment:
Again, I feel like this should just happen once in the `notify` handler?  Maybe I'm missing something.